tests/test_check.py

```diff
@@ -0,0 +1,391 @@
+UNPERMITTED_REPO = "opensafely/dummy_ons"
+NONEXISTENT_REPO = "opensafely/dummy"
+
+PERMITTED_HTTPS_ORIGIN = f"https://github.com/{PERMITTED_REPO}"
+PERMITTED_MULTIPLE_HTTPS_ORIGIN = (
+    f"https://github.com/{PERMITTED_MULTIPLE_REPO}"
+)
+UNPERMITTED_HTTPS_ORIGIN = f"https://github.com/{UNPERMITTED_REPO}"
+NONEXISTENT_HTTPS_ORIGIN = f"https://github.com/{NONEXISTENT_REPO}"
+
+PERMITTED_SSH_ORIGIN = f"git@github.com:{PERMITTED_REPO}.git"
+PERMITTED_MULTIPLE_SSH_ORIGIN = f"git@github.com:{PERMITTED_MULTIPLE_REPO}.git"
+UNPERMITTED_SSH_ORIGIN = f"git@github.com:{UNPERMITTED_REPO}.git"
+NONEXISTENT_SSH_ORIGIN = f"git@github.com:{NONEXISTENT_REPO}.git"
+
+
+STUDY_DEF_HEADER = """from cohortextractor import (StudyDefinition, patients)
+study = StudyDefinition ("""
+
+
+def write_restricted_files(path):
+    for a in [1, 2]:
+        with open(
+            os.path.join(str(path), f"study_definition_restricted_{a}.py"), "w"
+        ) as f:
+            f.write(f"{STUDY_DEF_HEADER}\n")
+            f.write(f"a={RESTRICTED_FUNCTION_CALL},\n")
+            f.write(f"#b={RESTRICTED_FUNCTION_CALL},\n")
+            f.write(f"c={UNRESTRICTED_FUNCTION_CALL},\n")
+            f.write(f"#d={UNRESTRICTED_FUNCTION_CALL},\n")
+            f.write(")")
+
+
+def write_unrestricted_files(path):
+    for a in [1, 2]:
+        with open(
+            os.path.join(str(path), f"study_definition_unrestricted_{a}.py"),
+            "w",
+        ) as f:
+            f.write(f"{STUDY_DEF_HEADER}\n")
+            f.write(f"#a={RESTRICTED_FUNCTION_CALL}\n")
+            f.write(f"b={UNRESTRICTED_FUNCTION_CALL}\n")
+            f.write(f"#c={UNRESTRICTED_FUNCTION_CALL}\n")
+            f.write(")")
+
+
+def git_init(url):
+    subprocess.run(["git", "init"])
+    subprocess.run(["git", "remote", "add", "origin", url])
+
+
+def validate_pass(capsys):
+    check.main()
+    stdout, stderr = capsys.readouterr()
+    assert stderr == ""
+    assert stdout == "Success\n"
+
+
+def validate_fail(capsys):
+    with pytest.raises(SystemExit):
+        check.main()
+        stdout, stderr = capsys.readouterr()
+        assert stdout != "Success\n"
+        assert stderr.startswith("Usage of restricted datasets found:")
+        assert "icnarc" in stderr
+        assert "admitted_to_icu" in stderr
+        assert "3:" in stderr
+        assert "4:" not in stderr
+        assert "5:" not in stderr
+        assert "#b=" not in stderr
+        assert "#d=" not in stderr
+        assert "unrestricted" not in stderr
+        assert "study_definition_restricted_1.py" in stderr
+        assert "study_definition_restricted_2.py" in stderr
+
+
+def test_unrestricted_local_norepo(tmp_path, capsys, monkeypatch):
+    if "GITHUB_REPOSITORY" in os.environ:
+        monkeypatch.delenv("GITHUB_REPOSITORY")
+    prev_dir = os.getcwd()
+    os.chdir(tmp_path)
+    write_unrestricted_files(tmp_path)
+    validate_pass(capsys)
+    os.chdir(prev_dir)
```

Comment:
ok, so this undoing won't work in the case of test failure.

I suggest a fixture:

```python
@pytest.fixture
def repo_dir(tmp_path):
    old = os.getcwd()
    os.chdir(tmp_path)
    yield tmp_path
    os.chdir(old)
```

And then use that instead of `tmp_path` fixture. That way, any test with `repo_dir` fixture will have the correct cleanup behaviour, and also you can use `repo_dir` exactly like `tmp_path`.